src/MatrixClient.ts

```diff
@@ -181,6 +182,34 @@ export class MatrixClient extends EventEmitter {
         return event;
     }
 
+    /**
+     * Gets the homeserver’s supported login types to authenticate users.
+     */
+    public getLoginFlows(): Promise<LoginFlows> {
+        return this.doRequest("GET", "/_matrix/client/r0/login");
+    }
+
+    /**
+     * Authenticate with the homeserver. This does NOT modify the access token of the client.
+     * @param type The type of login to attempt. Not all login types are supported by all homeservers, so use `getLoginFlows()` to check.
+     * @param identifier The identifier for the type of login being attempted. See https://spec.matrix.org/unstable/client-server-api/#identifier-types
+     * @param passwordOrToken The password for a `m.login.password` login attempt, or a token for `m.login.token`. Other `types` ignore this value.
+     * @param opts Extra device options to include.
+     */
+    public doLogin(type: LoginFlowsType, identifier: Record<string, string>, passwordOrToken?: string, opts?: {device_id?: string, initial_device_display_name?: string}): Promise<LoginResponse> {
```

Comment:
There should be nothing about login in the client class as it implies that the class can change users, which is not the case.